File: app/src/main/java/me/kirimin/mitsumine/ui/adapter/EntryInfoPagerAdapter.java
package me.kirimin.mitsumine.ui.adapter;

import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;

import java.util.ArrayList;
import java.util.List;

public class EntryInfoPagerAdapter extends FragmentPagerAdapter {

    private final List<Fragment> fragmentList = new ArrayList<>();
    private final List<String> titleList = new ArrayList<>();


    public EntryInfoPagerAdapter(FragmentManager fm) {
        super(fm);
    }

    @Override
    public Fragment getItem(int i) {
        return fragmentList.get(i);
    }

    @Override
    public int getCount() {
        return fragmentList.size();
    }

    @Override
    public CharSequence getPageTitle(int position) {
        return titleList.get(position);
    }

    public void addPage(Fragment fragment, String title) {
        fragmentList.add(fragment);
        titleList.add(title);
    }
}

File: app/src/main/java/me/kirimin/mitsumine/view/adapter/EntryInfoPagerAdapter.kt
package me.kirimin.mitsumine.view.adapter

import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentPagerAdapter

import java.util.ArrayList

public class EntryInfoPagerAdapter(fm: FragmentManager) : FragmentPagerAdapter(fm) {

    private val fragmentList = ArrayList<Fragment>()
    private val titleList = ArrayList<String>()

    override fun getItem(i: Int): Fragment {
        return fragmentList.get(i)
    }

    override fun getCount(): Int {
        return fragmentList.size()
    }

    override fun getPageTitle(position: Int): CharSequence {
        return titleList.get(position)
    }

    public fun addPage(fragment: Fragment, title: String) {
        fragmentList.add(fragment)
        titleList.add(title)
    }
}
